### ISSUE TITLE
RuntimeError "Could not parse document" on empty input in strict mode

### ISSUE THREAD

**Issue by semaperepelitsa** [sparklemotion/nokogiri#1349]
I noticed a change when updating from 1.5 to 1.6 probably related to https://github.com/sparklemotion/nokogiri/issues/1005

Previously, empty input would result in an empty document, and now it raises an error. However, the error is a generic RuntimeError instead of the expected Nokogiri::XML::SyntaxError. RuntimeError is very common in Ruby so I don't like to rescue it for this very specific case. Was it ever intended to be a RuntimeError?

Here is an example:

```ruby
gem "nokogiri", ARGV[0]
require "nokogiri"

begin
  p Nokogiri::XML(""){ |c| c.strict }
rescue => err
  p err
end

begin
  p Nokogiri::XML(" "){ |c| c.strict }
rescue => err
  p err
end
```

```
> ruby example.rb 1.6.6.2
#<RuntimeError: Could not parse document>
#<Nokogiri::XML::SyntaxError: Start tag expected, '<' not found>
> ruby example.rb 1.5.11
#<Nokogiri::XML::Document:0x3fdc498fe430 name="document">
#<Nokogiri::XML::SyntaxError: Start tag expected, '<' not found>
```

**Comment by flavorjones** [comment]
It's not due to #1005, as that was in v1.6.2.1 and that version emits a SyntaxError. I'm git bisecting now ...

**Comment by flavorjones** [comment]
Ah, wait, strike my last comment. I was misinterpreting the output of your script. Please hold ...

**Comment by flavorjones** [comment]
You're right, the offending commit is 2c9b7f1000b4870aa285e3c813f2de8c92593322 from #1005.

**Comment by flavorjones** [comment]
I've pushed a branch that addresses this, `flavorjones-gh-1349-syntax-error-on-empty-document`.

This will be in 1.7.0, as we're about to release 1.6.7 final. Leaving open until it's merged into master.

**Comment by semaperepelitsa** [comment]
Looks good. I will be waiting for the release. Thanks!

**Comment by flavorjones** [comment]
Merged onto master, will be in 1.6.8.
